### DISCUSSION TITLE
Getting warning while using mesh adaptivity

### DISCUSSION THREAD

**Discussion by avtarsinghh1991** [idaholab/moose#21213] [Q&A Meshing] [answered]
Hello MOOSE Experts

I am trying to implement the mesh adaptivity as follows:

```
[Adaptivity]
  marker = errorfrac 
  steps = 3
  [./Indicators]
    [./error]
      type = GradientJumpIndicator
      variable = damage
      outputs = none
    [../]
  [../]

  [./Markers]
  [./errorfrac]
      type = ValueThresholdMarker
      coarsen = 0
      variable = damage
      refine = 0.9
      outputs = none
    [../]
  [../]
[]
```

However, I am getting the following error:

```
Using EXPERIMENTAL Stateful Material Property projection with Adaptivity!

Warning! Mesh re-partitioning is disabled while using stateful material properties!  This can lead to large load imbalances and degraded performance!!
```

Can you please let me know what does it mean and how to avoid it?

Thanks in advance.

Best
Avtar

**Comment by GiudGiud** [comment, marked as answer]
Hello

The issue is that support for stateful properties (properties that depend on prior states) is growing but is not complete with regards to more advanced features like re-partitioning. 
See this issue for more details:
https://github.com/idaholab/moose/issues/4532

I think there's a more recent issue too but cant seem to find it right now

Guillaume

**Reply by avtarsinghh1991** [reply]
Hello @GiudGiud 
Thank you very much.
I think the only parameter is History field in phase field fracture.
Rest are all constants in my chemo-mechanical-fracture model.

Can you suggest any other alternative to use adaptive mesh?

Best
Avtar

**Reply by GiudGiud** [reply]
It's not barring you from using adaptivity here, it's just disabling re-partitioning which may lower performance if the load on each process becomes unequal.
I think the way forward is to take the performance hit until we fix this. It's on the schedule for next FY

**Reply by GiudGiud** [reply]
by performance hit I mean dont re-partition. Adaptivity should be fine though keep an eye out since it s experimental as mentioned in the message
